### ISSUE TITLE
Issues with dexterity collections and PTG

### ISSUE THREAD

**Issue by volkerjaenisch** [collective/collective.plonetruegallery#45]
I noticed some problems with dexterity collections from plone.app.contenttypes in Plone4.3 and Plone5. Namely they do not work anymore.
There are four problems:
1) The collections adapter is bound to plone.app.collections.interfaces.ICollection but dexterity collections have plone.app.contenttypes.interfaces.ICollection as marker interface. So the adapter does not fit the interface.
2) In dexterity the collections do no longer expose the getRawQuery function.
See https://community.plone.org/t/where-is-defined-collection-getrawquery-and-how/1040 .
This leads to a stack trace.
3) If this problem is overcome the wrong image type is used in the query.
4) And last but not least. The current code does not work with collections collecting subgalleries.

**Comment by volkerjaenisch** [comment]
Just install collective.plonetruegallery . Then replace the collection.py file that lives in this package with the code shown above.

**Comment by mactrash** [comment]
@volkerjaenisch 

Thanks for your reply, I am very new to this...

Do you mean after replay the file then I can use the gallery view in plone 5 collection?

**Comment by volkerjaenisch** [comment]
Yes. We are using PTG with collections in Plone 5. You only need to replace this one file.  If you got into problems feel free to ask.

**Comment by mactrash** [comment]
@volkerjaenisch I have follow the step you mentioned, buy the PTG gallery view still not show up in the selection list of my collections... any idea?

**Comment by volkerjaenisch** [comment]
The alternative is to use my fork of PTG:
https://github.com/volkerjaenisch/collective.plonetruegallery 
- Check out or download the repository to zeoserver/src/collective.plonetruegallery 
- Alter your buildout.cfg file: Add
  develop =
     src/collective.plonetruegallery
- run ./bin/buildout
- restart your plone

**Comment by mactrash** [comment]
@volkerjaenisch thanks for your help

I try your method and now with collective.plonetruegallery 3.4.7.dev0  but there is no gallery view in the collection ... do i missed something my plone is Plone 5.0.6rc1 (5016) do I need to active "Archetypes Content Types for Plone" ?

**Comment by mactrash** [comment]
@volkerjaenisch and my plone.app.collection  = 1.1.6
is that same as you using?

**Comment by volkerjaenisch** [comment]
The Version number 1.1.6 should not be an issue.
Please check the configuration for collections:
<your plone URL>/dexterity-types/Collection/@@behaviors

The entry:  
   Plone True Gallery
   Enable Plone True Gallery support
has to be enabled.

**Comment by volkerjaenisch** [comment]
Tested with plone.app.collection = 1.1.6, works flawlessly.

**Comment by mactrash** [comment]
@volkerjaenisch 

Thanks
Gallery view now show up in the collector and can be select but it got a error as below.

Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module collective.plonetruegallery.browser.views.galleryview, line 29, in __call__
  Module Products.Five.browser.pagetemplatefile, line 125, in __call__
  Module Products.Five.browser.pagetemplatefile, line 59, in __call__
  Module zope.pagetemplate.pagetemplate, line 132, in pt_render
  Module five.pt.engine, line 98, in __call__
  Module z3c.pt.pagetemplate, line 163, in render
  Module chameleon.zpt.template, line 261, in render
  Module chameleon.template, line 191, in render
  Module chameleon.template, line 171, in render
  Module d0e9d24677fac041b6f34863811f414d.py, line 459, in render
  Module e3bbe0aa01b667bbf022121054b66185.py, line 1223, in render_master
  Module e3bbe0aa01b667bbf022121054b66185.py, line 420, in render_content
  Module d0e9d24677fac041b6f34863811f414d.py, line 447, in __fill_content_core
  Module d0e9d24677fac041b6f34863811f414d.py, line 95, in render_content_core
  Module collective.plonetruegallery.galleryadapters.base, line 100, in number_of_images
  Module plone.memoize.instance, line 51, in memogetter
  Module collective.plonetruegallery.galleryadapters.basic, line 155, in cooked_images
  Module collective.plonetruegallery.galleryadapters.basic, line 142, in retrieve_images
  Module zope.component._api, line 109, in getMultiAdapter
ComponentLookupError: ((<Collection at /test/test>, <collective.plonetruegallery.galleryadapters.basic.BasicAdapter object at 0x7f775d31b1d0>), <InterfaceClass collective.plonetruegallery.interfaces.IImageInformationRetriever>, u'')

 - Expression: "python: view.adapter.number_of_images == 0"
 - Filename:   ... ctive/plonetruegallery/browser/views/templates/layout.pt
 - Location:   (line 22: col 45)
 - Source:     ... :condition="python: view.adapter.number_of_images == 0">
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Arguments:  repeat: {...} (0)
               template: <ViewPageTemplateFile - at 0x7f775c3ee810>
               views: <ViewMapper - at 0x7f775d31ba90>
               modules: <instance - at 0x24c6128>
               args: <tuple - at 0x7f7770417050>
               here: <ImplicitAcquisitionWrapper test at 0x917b690>
               user: <ImplicitAcquisitionWrapper - at 0x917bb40>
               loop: {...} (0)
               nothing: <NoneType - at 0x783160>
               container: <ImplicitAcquisitionWrapper test at 0x917b690>
               request: <instance - at 0x7f775d961200>
               wrapped_repeat: <SafeMapping - at 0x7f775d892470>
               traverse_subpath: <list - at 0x7f775d95b098>
               default: <object - at 0x7f77703dac00>
               context: <ImplicitAcquisitionWrapper test at 0x917b690>
               view: <SimpleViewClass from /opt/plone/buildout-cache/eggs/collective.plonetruegallery-3.4.7-py2.7.egg/collective/plonetruegallery/browser/views/templates/layout.pt galleryview at 0x7f775c959cd0>
               translate: <function translate at 0x7f775c937140>
               root: <ImplicitAcquisitionWrapper Zope at 0x7463d20>
               options: {...} (0)
               target_language: <NoneType - at 0x783160>

**Comment by volkerjaenisch** [comment]
Are you sure that you have only images (Type Image) and No PDF-Files etc. in the collection?
Have you any Images in the collection?

**Comment by mactrash** [comment]
Yes, 1 jpg can show with Album view.

**Comment by volkerjaenisch** [comment]
Can you please make sure that the alternative collectiv.plonetruegallery is really used:
-  Deinstall the collectiv.plonetruegallery package via <your plone url>/prefs_install_products_form
- Stop Plone
- Make sure there is no folder zeocluster/../buildout-cache/eggs/collective.plonetruegallery
- run bin/buildout
- Make sure there is no folder zeocluster/../buildout-cache/eggs/collective.plonetruegallery
    If there is your buildout is not correct. Post your buildout.cfg here
- If there is no such folder
- Start Plone
- Install the collectiv.plonetruegallery package via <your plone url>/prefs_install_products_form

**Comment by mactrash** [comment]
Ok here is the result,

I don't know why it error and not build

 sudo -u plone_buildout bin/buildout
Develop: '/opt/plone/zinstance/src/collective.plonetruegallery'
warning: no files found matching '*.rst'
warning: no previously-included files found matching '.coveragerc'
warning: no previously-included files found matching 'buildout.cfg'
warning: no previously-included files found matching 'tox.ini'
warning: no previously-included files found matching 'travis.cfg'
warning: no previously-included files matching '*.pyc' found anywhere in distribution
Uninstalling precompiler.
Uninstalling zopepy.
Uninstalling instance.

**Comment by volkerjaenisch** [comment]
You have to enter the virtual environment first.
in zeoserver dir:  
source bin/activate
./bin/buildout 

If this fails please post the complete output

**Comment by mactrash** [comment]
@volkerjaenisch 

It is working now, thanks alot.

**Comment by volkerjaenisch** [comment]
you are welcome
